scipy/sparse/construct.py

```diff
@@ -738,12 +742,14 @@ def random(m, n, density=0.01, format='coo', dtype=None,
     --------
     >>> from scipy.sparse import random
     >>> from scipy import stats
+    >>> import numpy as np
+    >>> from numpy.random import default_rng
+    >>> rng = default_rng()
 
-    >>> class CustomRandomState(np.random.RandomState):
+    >>> class CustomRandomState(np.random.Generator):
```

Comment:
How necessary is this subclassing to the example? It's not commented on, or described as an intended API elsewhere in the docstring. Changing the semantics of a method like this is not a practice that I would recommend. If one needs to be able to specify to this method to sample even integers, as in this example, that needs to be specified by some other means.

Also, just swapping `Generator` for `RandomState` won't work in this example. This function uses the `rng_integers()` compatibility shim to draw integers. If `random_state` is a `Generator`, it will call the `integers()` method directly instead of `randint()`, so this overridden method gets ignored.

Reply:
Indeed, I don't think this class is needed in the example. I will remove it.